### PR TITLE
Fix getFlattenedItemList type in index.d.ts

### PR DESCRIPTION
**Pull request by 1amageek**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.

Title: [apple-mapkit-js-browser] Fix type definition for getFlattenedItemList

- [x] Test the change in your own code. (Compile and run.)

- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.

- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).

- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).

- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://developer.apple.com/documentation/mapkitjs/itemcollection/1452582-getflatteneditemlist

- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

According to the Apple MapKit JS documentation, `getFlattenedItemList` is a function that returns an array of annotations and overlays, not a property directly holding the array. This PR fixes the type definition to match the actual API.

https://developer.apple.com/documentation/mapkitjs/itemcollection

### CODE REVIEW

**Review by MysteryBlokHed — COMMENTED** [review]
If you update the tests ([apple-mapkit-js-browser-tests.ts](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/apple-mapkit-js-browser/apple-mapkit-js-browser-tests.ts)), then this can be merged without approval from a DefinitelyTyped maintainer.

**Review by andrewbranch — CHANGES_REQUESTED** [review]
The docs say

```
attribute (mapkit.Annotation|mapkit.Overlay)[] getFlattenedItemList;
```

which is the direct array, not a function returning it. The name does sound like it would be a function, but the type annotation is clear. You’ll have to prove that the Apple docs are wrong and this PR is correct to move forward.

Screenshot since the link seems to have an unreliable id in it:

image

https://developer.apple.com/documentation/mapkitjs/itemcollection/2978150-getflatteneditemlist

**Pull request closed without merging**
